docs/modules/ROOT/pages/how-tos/testing/build/index.adoc

```diff
@@ -1 +1,56 @@
-= Build-time tests
\ No newline at end of file
+= Build-time tests
+
+This document covers the build-time tests that {ProductName} runs as part of its component build pipeline. These build-time tests automatically check all application images to inform you if they're up-to-date, correctly formatted, and protected from security vulnerabilities.
+
+NOTE: These tests are non-blocking and do not prevent the pipeline from proceeding if issues are found.
+
+The component build pipeline for {ProductName} include various types of build-time tests, conducted using link:https://tekton.dev/docs/pipelines/tasks/#overview[Tekton tasks]. For validating container information, the pipeline employs https://www.conftest.dev/[Conftest]. 
+
+The table below outlines the default build-time tests:
+
+.Deprecated image checks
+|===
+|Test name |Description |Failure message
+
+|image_repository_deprecated |Deprecated images are no longer maintained, leading to unresolved security vulnerabilities. | The container image must not be built from a repository  marked as 'Deprecated' in COMET
+|===
+
+.Unsigned RPM check
+|===
+|Test name |Description |Failure message
+
+|image_unsigned_rpms |Packages signed with Red Hat's secure signing server adheres to stringent policies and procedures. |Alerts the user for any unsigned RPMs. Found following unsigned rpms(nvra):
+|===
+
+.Source code check
+|===
+|Test name |Description |Failure message
+
+|sast-snyk-check |Scans and analyze source code or compiled versions of code to help find security flaws. | <tbd>
+|===
+
+You can disable the default build-time tests by setting the `skip-checks` parameter to `true`
+
+[source,yaml]
+----
+- description: Skip checks against built image
+  name: skip-checks
+  type: string
+  default: "false"
+----
+
+.Security checks
+|===
+|Test name |Description |Failure message
+
+|clair-scan |Scans container images for vulnerabilities using Clair, by comparing the components of container image against Clair's vulnerability databases. | Found packages with critical vulnerabilities. Consider updating to a newer version of those packages, they may no longer be affected by the reported CVEs.
+
+|clamav-scan |Scans the content of container images for viruses, malware, and other malicious content using ClamAV antivirus scanner. | A malware has been found.
+|===
```

Comment:
This PR only adds this part ([see](https://github.com/gtrivedi88/konflux-docs/compare/HACDOCS-924...dheerajodha:docs:STONEINTG-766?expand=1)), and everything else is part of another PR so please ignore it.

Reply:
Shall we mention also snyk task overview in this section?

Reply:
I was thinking about it but Gaurav added info on Snyk just a few lines above this section ([here](https://github.com/konflux-ci/docs/pull/47/files#diff-874b378b08d8bdd02bf106c5357653fc2d4c2a0a4bb839bfeec712fe342f7647R29)), wdyt?

Reply:
SAST = Static application security testing, it has security in name, I think it should be within this section

Reply:
ack, shifted it to this section but I have no clue what to put under the "Failure message" column :neutral_face:

Reply:
cc: @gtrivedi88 I've moved the Snyk task within this new "Security checks" table.